invoices/invoice_expiry_watcher.go

```diff
@@ -14,6 +14,12 @@ import (
 	"github.com/lightningnetwork/lnd/zpay32"
 )
 
+const (
+	// watcherChanSize defines the size of the InvoiceExpiryWatcher's
+	// incoming invoices channel.
+	watcherChanSize = 5
```

Comment:
How does this guarantee that the dead lock cannot happen anymore?

Reply:
Also realized while rethinking this quick fix that it doesn't help when the channel is filled faster than consumed. I'll  prepare a better fix that makes the deadlock impossible.

Reply:
ptal